src/main/java/hudson/plugins/sshslaves/JavaVersionChecker.java

```diff
@@ -88,6 +88,10 @@ private String checkJavaVersion(TaskListener listener, String javaCommand) throw
         StringWriter output = new StringWriter();   // record output from Java
 
         ByteArrayOutputStream out = new ByteArrayOutputStream();
+        // if the command has a space, we need to quote it
+        if(javaCommand.contains(" ")) {
+            javaCommand = "\"" + javaCommand + "\"";
+        }
         connection.exec(javaCommand + " "+ jvmOptions + " -version",out);
```

Comment:
another option is to put the Java path between quotes it should work on windows and Unix

```suggestion
        connection.exec("\"" + javaCommand + "\" "+ jvmOptions + " -version",out);
```